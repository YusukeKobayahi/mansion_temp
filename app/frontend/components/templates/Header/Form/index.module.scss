@import "~/styles/utils";
@import "~/styles/variables";

.form {
  width: 260px;
  position: relative;
  height: 36px;

  @include screenMd {
    width: 100%;
    height: 46px;
  }

  input {
    width: 100%;
    height: 100%;
    padding: 0 15px;
    border: 1px solid #c3ae75;
    border-radius: 50px;
    font-size: 16px;
    box-sizing: border-box;

    @include screenMd {
      border-radius: 0;
      border: none;
    }

    &::placeholder {
      font-size: 14px;
    }
  }

  .submit {
    position: absolute;
    background-color: unset;
    padding: 0;
    right: 10px;
    top: 55%;
    transform: translateY(-50%);
  }
}
