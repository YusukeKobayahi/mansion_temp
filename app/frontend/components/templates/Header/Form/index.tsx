import { useState } from "react";
import Image from "next/image";
import styles from "./index.module.scss";

const Form: React.FC = () => {
  const [word, setWord] = useState("");
  const actionUrl = "/ms-library/freeword/";
  return (
    <form
      action={actionUrl}
      method="GET"
      className={styles.form}
      onSubmit={(e) => {
        if (word.length < 2) e.preventDefault();
      }}
    >
      <input
        name="wd"
        type="text"
        value={word}
        onChange={(e) => setWord(e.target.value)}
        placeholder="マンション名を入力してください"
      />
      <button className={styles.submit} type="submit">
        <Image
          className={styles.search}
          src={"/ms-library/images/search.svg"}
          alt={"housii"}
          width={18}
          height={18}
          loading={"eager"}
        />
      </button>
    </form>
  );
};

export default Form;
