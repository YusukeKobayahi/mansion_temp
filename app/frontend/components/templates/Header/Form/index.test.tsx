import { render } from "@testing-library/react";
import Form from "~/components/templates/Header/Form";

describe("components/commons/Header/Form", () => {
  it("正常に表示される", () => {
    const { asFragment } = render(<Form />);
    expect(asFragment()).toMatchSnapshot();
  });
});
