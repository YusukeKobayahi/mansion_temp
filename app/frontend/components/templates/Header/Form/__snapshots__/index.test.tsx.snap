// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/commons/Header/Form 正常に表示される 1`] = `
<DocumentFragment>
  <form
    action="/ms-library/freeword/"
    class="form"
    method="GET"
  >
    <input
      name="wd"
      placeholder="マンション名を入力してください"
      type="text"
      value=""
    />
    <button
      class="submit"
      type="submit"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTgiIGhlaWdodD0iMTgiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <img
          alt="housii"
          class="search"
          data-nimg="true"
          decoding="async"
          src="/_next/image?url=%2Fms-library%2Fimages%2Fsearch.svg&w=48&q=75"
          srcset="/_next/image?url=%2Fms-library%2Fimages%2Fsearch.svg&w=32&q=75 1x, /_next/image?url=%2Fms-library%2Fimages%2Fsearch.svg&w=48&q=75 2x"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </button>
  </form>
</DocumentFragment>
`;
